packages/javascript-decorators/README.md

```diff
@@ -0,0 +1,117 @@
+# `@shopify/javascript-decorators`
+
+[![Build Status](https://travis-ci.org/Shopify/quilt.svg?branch=master)](https://travis-ci.org/Shopify/quilt)
+[![License: MIT](https://img.shields.io/badge/License-MIT-green.svg)](LICENSE.md) [![npm version](https://badge.fury.io/js/%40shopify%2Fjavascript-decorators.svg)](https://badge.fury.io/js/%40shopify%2Fjavascript-decorators.svg) [![npm bundle size (minified + gzip)](https://img.shields.io/bundlephobia/minzip/@shopify/javascript-decorators.svg)](https://img.shields.io/bundlephobia/minzip/@shopify/javascript-decorators.svg)
+
+A set of decorators to aid your javascript journey.
+
+## Installation
+
+```bash
+$ yarn add @shopify/javascript-decorators
+```
+
+## Usage
+
+### `memoize`
+
+The memoize decorator creates a function that memoizes the results of the function it is decorating.
+The cache key for storing the results are based on the first argument provided to the memoized function.
+If the memoization key should not be inferred from the first argument alone, a `resolver` should be passed in to ensure a unique key.
+
+Know that memoization will be skipped on server process and the cached results have a maximum limit of 50 entries on a first in first out basis.
+
+#### Memoizing a simple function
+
+```ts
+import {memoize} from '@shopify/javascript-decorators';
+
+class MyClass {
+  ...
+  @memoize()
+  addOne(number: number) {
+    return number + 1;
+  }
+  ...
+}
+
+const myClass = new MyClass();
+
+myClass.addOne(1); // -> 2, addOne is executed
+myClass.addOne(1); // -> 2, result is from cache
+```
+
+#### Memoizing a function with object as argument
+
+When memoizing a function with object as first argument, make sure the object is immutable.
+
+```ts
+import {memoize} from '@shopify/javascript-decorators';
+
+class MyClass {
+  ...
+  @memoize()
+  getValues(someObject: {id: string, value: string}) {
+    return Object.values(someObject);
+  }
+  ...
+}
+
+const myClass = new MyClass();
+
+const testObject1 = {id: 1, value: 2};
+myClass.getValues(testObject1); // -> [1, 2], getValues is executed
+myClass.getValues(testObject1); // -> [1, 2], result is from cache
+
+testObject1.value = 3;
+myClass.getValues(testObject1); // -> [1, 2], result is from cache, BAD
+```
+
+#### Memoizing a function with resolver
+
+The resolver takes in the same arguments as the function it is decorating.
+Be sure that the resolver returns a unique identifer.
+
+This is how you can use a resolver to make sure the results of the function will always be correct.
```

Comment:
This feels like all one paragraph. I am not really sure this last sentence makes sense though, it isn't really about them being "correct", it is about getting a memoized value for an object that changes from call to call